bundle/bundle.go

```diff
@@ -268,6 +289,21 @@ func (m *Manifest) validateAndInjectDefaults(b Bundle) error {
 		wasmModuleToEps[wmConfig.Module] = wmConfig.Entrypoint
 	}
 
+	// Validate data patches in bundle.
+	for _, patch := range b.Patch.Data {
+		path := strings.Trim(patch.Path, "/")
+		found := false
+		for i := range roots {
+			if strings.HasPrefix(path, roots[i]) {
```

Comment:
This came up in the last review too. See [this](https://github.com/open-policy-agent/opa/pull/3456#discussion_r634624808) for context.

Reply:
I think we should fix this to compare path segments correctly. Thinking about this a bit more, if we change the validation to compare path segments as opposed to strings, that shouldn't break anything because the validation will be less strict, right?

Reply:
With the prefix comparison the following would be a valid path
```
patch path = /foo/bar
root path = /foo/ba
```

If we update to path segment comparison, it would be an invalid patch path. This seems to be the behavior we want.

Reply:
Updated to use path segment comparison for the manifest root check, let me know what y'all think.